src/main/java/macrobase/analysis/pipeline/OneShotPipeline.java

```diff
@@ -0,0 +1,36 @@
+package macrobase.analysis.pipeline;
+
+import macrobase.analysis.result.AnalysisResult;
+import macrobase.conf.ConfigurationException;
+import macrobase.conf.MacroBaseConf;
+
+import java.io.IOException;
+import java.sql.SQLException;
+
+
+abstract public class OneShotPipeline extends AbstractPipeline {
```

Comment:
Agreed. But, we had a dilemma how to handle returning of results during the stream processing. Ideally we would want to return results multiple times, but we don't have user interface to support that.. also we are not sure if we should return updated results or the results about new data points.. So this seemed like a solution that works for now.